src/Dapr.AspNetCore/DaprEndpointRouteBuilderExtensions.cs

```diff
@@ -84,12 +86,52 @@ private static IEndpointConventionBuilder CreateSubscribeEndPoint(IEndpointRoute
                         .Select(part => part.Content))));
 
                     var rawPayload = entry.EnableRawPayload ?? options?.EnableRawPayload;
-                    var subscription = new Subscription
+                    var subscriptionKey = entry.PubsubName + ":" + entry.Name;
+                    Subscription subscription;
+                    try
                     {
-                        Topic = entry.Name,
-                        PubsubName = entry.PubsubName,
-                        Route = route
-                    };
+                        subscription = subscriptionMap[subscriptionKey];
+                    } catch (KeyNotFoundException) {
+                        subscription = new Subscription
+                        {
+                            Topic = entry.Name,
+                            PubsubName = entry.PubsubName,
+                        };
+                        subscriptions.Add(subscription);
+                        subscriptionMap[subscriptionKey] = subscription;
+                    }
+
+                    if (string.IsNullOrEmpty(entry.Match))
+                    {
+                        if (subscription.Routes != null &&
+                            subscription.Routes.Rules.Count > 0)
+                        {
+                            subscription.Routes.Default = route;
+                        } else
+                        {
+                            subscription.Route = route;
```

Comment:
It seems like the logic here depends on a lot on what order we see the attributes. Would it simplify things to remove the `.SelectMany` on line 65 and process a group at a time. 

I'm particularly passionate about doing a good job reporting invalid configurations. Users frequently ask for help troubleshooting pub/sub configurations.